dll/win32/shell32/folders/CFSFolder.cpp

```diff
@@ -14,19 +14,43 @@ WINE_DEFAULT_DEBUG_CHANNEL (shell);
 
 static HRESULT SHELL32_GetCLSIDForDirectory(LPCWSTR pwszDir, LPCWSTR KeyName, CLSID* pclsidFolder);
 
+static BOOL ItemIsFolder(PCUITEMID_CHILD pidl)
+{
+    const BYTE mask = PT_FS | PT_FS_FOLDER_FLAG | PT_FS_FILE_FLAG;
+    const BYTE type = _ILGetType(pidl);
+    return (type & mask) == (PT_FS | PT_FS_FOLDER_FLAG) || (type == PT_FS && ILGetNext(pidl));
+}
+
 static LPCWSTR GetItemFileName(PCUITEMID_CHILD pidl, LPWSTR Buf, UINT cchMax)
 {
     FileStructW* pDataW = _ILGetFileStructW(pidl);
     if (pDataW)
         return pDataW->wszName;
     LPPIDLDATA pdata = _ILGetDataPointer(pidl);
-    if ((pdata->type & PT_VALUEW) == PT_VALUEW)
+    if (pdata && (pdata->type & PT_VALUEW) == PT_VALUEW) // TODO: || PT_FOLDERW
```

Comment:
I suppose the todo is for a future PR perhaps?

Reply:
I did not want to change this now but yes, PT_FOLDERW stores its string in the exact same place as PT_VALUEW.